src/component/variantPage/CheckBoxContainer.tsx

```diff
@@ -0,0 +1,103 @@
+import autobind from "autobind-decorator";
+import * as React from 'react';
+import {FormEvent} from "react";
+import {
+    Form, Row, Col
+} from "react-bootstrap";
+import CheckBox from "./CheckBox";
+import { observable, action, computed } from "mobx";
+import { toggleIncluded } from "../../lib/ArrayUtils";
+import { observer } from "mobx-react";
+import "./CheckboxContainer.css";
+import { VariantStore } from "../../page/VariantStore";
+
+interface ICheckContainerProps
+{
+    store: VariantStore;
+    allCheckboxNames: string[];
+}
+
+@observer
+class CheckBoxContainer extends React.Component<ICheckContainerProps>
+{
+    public render()
+    {
+        return (
+            <div>
+                <Form>
+                    <Row style={{fontSize: "1rem"}}>
+                        <Col lg="6" className="justify-content-center d-flex mb-4">
+                            <CheckBox 
+                                key={"Select all"}
+                                checkboxName={"Select all"}
+                                isChecked={this.isAllSelected}
+                                onChange={this.onSelectAll}
+                                className={"NoColor"}/>
+                        </Col>
+                        <Col lg="6" className="justify-content-center d-flex mb-4">
+                            <CheckBox 
+                                key={"Remove all"}
+                                checkboxName={"Remove all"}
+                                isChecked={this.isAllRemoved}
+                                onChange={this.onRemoveAll}
+                                className={"NoColor"}/>
+                        </Col>
+                    </Row>
+                    <div>
+                        {this.props.allCheckboxNames.map((name) => {
+                            return (
+                                <Row>
+                                    <Col>
+                                        <CheckBox key={name} checkboxName={name} isChecked={this.selectedCheckboxNames.includes(name)} onChange={this.onSelectionChange}/>
+                                    </Col>
+                                </Row>
+                            )
+                        })}
+                    </div>
+                </Form>
+            </div>
+        );
+    }
+
+    @observable
+    selectedCheckboxNames = this.props.allCheckboxNames;
```

Comment:
This should probably be a `@computed` instead of an `@observable`. You can use `this.props.store.selectedRecources` to compute this one.